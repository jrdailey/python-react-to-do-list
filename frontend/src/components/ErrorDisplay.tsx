interface ErrorDisplayProps {
  errorMessage: string,
}

const ErrorDisplay = ({
  errorMessage,
}: ErrorDisplayProps) => {
  return (
    <div role="alert" className="my-2 p-2 font-semibold text-white text-center rounded-md bg-red-400 border-2 border-red-800">
      {errorMessage}
    </div>
  )
}

export default ErrorDisplay
