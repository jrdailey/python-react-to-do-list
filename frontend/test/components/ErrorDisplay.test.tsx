import ErrorDisplay from '../../src/components/ErrorDisplay'
import { render } from '@testing-library/react'

describe('ErrorDisplay', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<ErrorDisplay errorMessage='Something went wrong' />)

    expect(asFragment()).toMatchSnapshot()
  })
})
