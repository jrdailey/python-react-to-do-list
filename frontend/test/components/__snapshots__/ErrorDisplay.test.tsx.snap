// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorDisplay > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-2/3 my-2 p-2 font-semibold text-white text-center rounded-md bg-red-400 border-2 border-red-800"
  >
    Something went wrong
  </div>
</DocumentFragment>
`;
